### PR TITLE
Adding a consistent way to select the tags to be applied to build/push

### PR DESCRIPTION
**Pull request by jvanzyl**

### CODE REVIEW

File: plugin/src/test/java/com/spotify/plugin/dockerfile/TagsSelectorTest.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    List<String> tags;
+
+    // Build
+    //
+    // <tags>
+    //   <tag>latest</tag>
+    // </tags>
+    //
+    tags = selector.select(ImmutableList.of("latest"), null, null);
+    assertThat(tags.size()).isEqualTo(1);
```
Comment (losipiuk) [inline plugin/src/test/java/com/spotify/plugin/dockerfile/TagsSelectorTest.java]: nit: you could use https://joel-costigliola.github.io/assertj/core-8/api/org/assertj/core/api/ListAssert.html#containsExactly-ELEMENT...-

File: plugin/src/test/java/com/spotify/plugin/dockerfile/TagsSelectorTest.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+
+import com.google.common.collect.ImmutableList;
+import java.util.List;
+import org.junit.Test;
+
+public class TagsSelectorTest {
+
+  @Test
+  public void validateTagsAreSelectedCorrectly() {
```
Comment (losipiuk) [inline plugin/src/test/java/com/spotify/plugin/dockerfile/TagsSelectorTest.java:33]: It seems there is not test with no metadata and no explicit tags which should default to "latest"

File: plugin/src/main/java/com/spotify/plugin/dockerfile/TagsSelector.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+      // 3) use latest
+      //
+      if (tag != null) {
+        tagsToUse.add(tag);
+      } else {
+        // Metadata is written by the BuildMojo and this will only be picked up by the PushMojo
+        if (tagFromMetadata != null) {
+          tag = tagFromMetadata;
+        } else {
+          tag = "latest";
```
Comment (losipiuk) [inline plugin/src/main/java/com/spotify/plugin/dockerfile/TagsSelector.java:55]: nit: extract constant. Maybe there is one already?

**Review by losipiuk — APPROVED** [review]

**Comment by losipiuk** [comment]
thx

File: plugin/src/test/java/com/spotify/plugin/dockerfile/TagsSelectorTest.java (hunk 1 above)
Comment (jvanzyl) [reply plugin/src/test/java/com/spotify/plugin/dockerfile/TagsSelectorTest.java]: Yes, that's nicer.

File: plugin/src/test/java/com/spotify/plugin/dockerfile/TagsSelectorTest.java (hunk 2 above)
Comment (jvanzyl) [reply plugin/src/test/java/com/spotify/plugin/dockerfile/TagsSelectorTest.java:33]: Added

File: plugin/src/main/java/com/spotify/plugin/dockerfile/TagsSelector.java (hunk 3 above)
Comment (jvanzyl) [reply plugin/src/main/java/com/spotify/plugin/dockerfile/TagsSelector.java:55]: Added

**Pull request merged**